File: lib/bodies/category.dart
import 'package:mango_ui/keys.dart';

class Category {
  final String name;
  final String text;
  final String description;
  final String pageUrl;
  final num baseCategory;
  final String clientID;
  final Key imageKey;
  final Key ownerKey;

  Category(this.name, this.text, this.description, this.pageUrl,
      this.baseCategory, this.clientID, this.imageKey, this.ownerKey);

  Map<String, dynamic> toJson() {
    return {
      "Name": name,
      "Text": text,
      "Description": description,
      "PageURL": pageUrl,
      "BaseCategory": baseCategory,
      "ClientID": clientID,
      "ImageKey": imageKey.toJson(),
      "OwnerKey": ownerKey.toJson(),
    };
  }
}

File: core/category.go
package core

import (
	"github.com/louisevanderlith/husk/hsk"
	"github.com/louisevanderlith/husk/keys"
	"github.com/louisevanderlith/husk/validation"
	"github.com/louisevanderlith/stock/core/categories"
)

type Category struct {
	Name         string          `hsk:"size(128)"`
	Text         string          `hsk:"size(256)"`
	Description  string          `hsk:"size(512)"`
	PageURL      string          `hsk:"size(128)"`
	BaseCategory categories.Enum `hsk:"null"`
	ClientID     string
	ImageKey     *keys.TimeKey
	OwnerKey     hsk.Key
}

func (c Category) Valid() error {
	return validation.Struct(c)
}
